### PR TITLE
Add rbac and psp

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards [#2296](https://github.com/giantswarm/giantswarm/issues/2296)

Add RBAC for the KVM operator based on the k8s API calls found in the project and dependencies

### CODE REVIEW

**Comment by kopiczko** [comment]
Did you test that? I think flannel-operator will require the same.

**Comment by pipo02mix** [comment]
No, I am waiting for having an environment ready. Vol is working on it.

Flannel and endpoint operators are WIP

File: helm/kvm-operator-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    resources:
+      - endpoints
+    verbs:
+      - list
+      - watch
+      - create
+  - apiGroups:
+      - ""
+    resources:
+      - persistenvolumeclaims
```
Comment (puja108) [inline helm/kvm-operator-chart/templates/rbac.yaml]: typo?

File: helm/kvm-operator-chart/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      - delete
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+      - secrets
+      - resourcequotas
+      - limitranges
+      - nodes
+      - persistenvolumes
```
Comment (puja108) [inline helm/kvm-operator-chart/templates/rbac.yaml]: typo?

File: helm/kvm-operator-chart/templates/rbac.yaml (hunk 1 above)
Comment (pipo02mix) [reply helm/kvm-operator-chart/templates/rbac.yaml]: yes thanks

**Comment by pipo02mix** [comment]
First of all, say this PR has converted into a monster unintentionally. I started thinking only RBAC/PSP specs were needed but then I discovered guest clusters are generated in different namespaces so service account and bindings should be created at runtime for this specific cluster. After adding service account and cluster role binding into resource folders (v2 and v3) I don't know what is left to let operator kit realized about these resources and manage them properly. What do I should do to configure it?

cc/ @xh3b4sd @kopiczko

File: service/keyv2/key.go [hunk 3]
```diff
@@ -36,7 +36,11 @@ const (
 
 	FlannelEnvPathPrefix = "/run/flannel"
 
-	K8SKVMHealthDocker = "quay.io/giantswarm/k8s-kvm-health:ddf211dfed52086ade32ab8c45e44eb0273319ef"
+	K8SKVMHealthDocker       = "quay.io/giantswarm/k8s-kvm-health:ddf211dfed52086ade32ab8c45e44eb0273319ef"
+	K8SEndpointUpdaterDocker = "quay.io/giantswarm/k8s-endpoint-updater:df982fc73b71e60fc70a7444c068b52441ddb30e"
+	// kvm endpoint annotations
+	AnnotationIp      = "endpoint.kvm.giantswarm.io/ip"
+	AnnotationService = "endpoint.kvm.giantswarm.io/service"
```
Comment (xh3b4sd) [inline service/keyv2/key.go]: These changes are already in master. I wonder why this is shown as added. Are you doing some weird force pushes? This looks dangerously wrong to me personally.

Comment (pipo02mix) [reply service/keyv2/key.go]: no I have just rebased my branch

**Comment by pipo02mix** [comment]
ok let me try split my work in different PRs

**Comment by pipo02mix** [comment]
Closing this in favor of the smaller PRs

**Pull request closed without merging**